sdks/go/pkg/beam/util/harnessopts/sampler.go

```diff
@@ -0,0 +1,38 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package harnessopts defines user-facing entrypoints into Beam hooks
+// affecting the SDK harness. Call these functions at any time before
+// submitting your pipeline to a runner, for that pipeline's workers to be affected.
+package harnessopts
+
+import (
+	"strconv"
+	"time"
+
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/core/util/hooks"
+)
+
+const (
+	samplePeriodHook = "beam:go:hook:dofnmetrics:sampletime"
+)
+
+// SampleInterval sets the sampling time period for DoFn metrics sampling.
+// Default value is 200ms.
+func SampleInterval(samplePeriod time.Duration) error {
+	sampleTime := strconv.FormatInt(int64(samplePeriod), 10)
```

Comment:
Instead of designing your own format, consider just using the built in conversions for time.Duration to string.
eg. the String representation of Duration, and `time.ParseDuration`
https://play.golang.org/p/RWuZzIkwHD0

So this side would use samplePeriod.String()